addons/l10n_ro_edi/models/account_edi_xml_ubl_ciusro.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, _
+
+
+SECTOR_RO_CODES = ('SECTOR1', 'SECTOR2', 'SECTOR3', 'SECTOR4', 'SECTOR5', 'SECTOR6')
+
+
+class AccountEdiXmlUBLRO(models.AbstractModel):
+    _inherit = "account.edi.xml.ubl_bis3"
+    _name = "account.edi.xml.ubl_ro"
+    _description = "CIUS RO"
+
+    def _export_invoice_filename(self, invoice):
+        return f"{invoice.name.replace('/', '_')}_cius_ro.xml"
+
+    def _get_partner_address_vals(self, partner):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals = super()._get_partner_address_vals(partner)
+
+        if partner.state_id:
+            vals["country_subentity"] = partner.country_code + '-' + partner.state_id.code
+
+        return vals
+
+    def _get_invoice_tax_totals_vals_list(self, invoice, taxes_vals):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals_list = super()._get_invoice_tax_totals_vals_list(invoice, taxes_vals)
+
+        if invoice.currency_id.name != 'RON':
+            # add new TaxTotal in RON
+            from_currency = invoice.currency_id
+            ron_currency = self.env.ref("base.RON")
+            from_amount = vals_list[0]['tax_amount']
+            to_amount = from_currency._convert(from_amount, ron_currency, invoice.company_id, invoice.date, False)
+
+            vals_list.append({
+                'currency': ron_currency,
+                'currency_dp': ron_currency.decimal_places,
+                'tax_amount': to_amount,
+            })
+
+        return vals_list
+
+    def _get_partner_party_tax_scheme_vals_list(self, partner, role):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals_list = super()._get_partner_party_tax_scheme_vals_list(partner, role)
+
+        if not partner.vat and partner.company_registry:
+            company_registry = partner.company_registry
+            if not company_registry.startswith(partner.country_code):
+                company_registry = partner.country_code + company_registry
```

Comment:
You could add some constraints on the company_registry in `_export_invoice_constraints` to avoid doing this check :)
FYI, we didn't do it in `account_edi_ubl_cii` because before saas-16.2, the constraints weren't blocking.